pgjdbc/src/test/java/org/postgresql/test/jdbc2/StatementTest.java

```diff
@@ -411,6 +421,104 @@ public void testWarningsAreCleared() throws SQLException {
     stmt.close();
   }
 
+  @Test
+  public void testWarningsAreAvailableAsap()
+      throws SQLException, InterruptedException, ExecutionException {
+    con.createStatement()
+            .execute("CREATE OR REPLACE FUNCTION notify_then_sleep() RETURNS VOID AS "
+                + "$BODY$ "
+                + "BEGIN "
+                + "RAISE NOTICE 'Test 1'; "
+                + "RAISE NOTICE 'Test 2'; "
+                + "EXECUTE pg_sleep(2); "
```

Comment:
Great idea, I have now implemented this, it should speed up the test a bit and make it a little more robust.